### PR TITLE
Add `mean_squared_error` functional metric

### LINKED ISSUE #512: Add `mean_squared_error` functional metric
### 🚀 Task description

Currently, we use `sklearn.metrics.mean_squared_error` in `etna.metrics.MSE`, but it can't handle missing values.

We are going to add our own implementation of `mean_squared_error` to work around missing values.

### Plan

- Add `mean_squared_error` into `etna.metrics.functional_metrics`
  - Ignoring missing values could be done by using `np.nanmean` instead of `np.average` in implementation of `sklearn.metrics.mean_squared_error`
  - NaNs should be ignored in both `y_true` and `y_pred`
  - If all values are NaNs the result value should be None
- Use new `mean_squared_error` in `etna.metrics.MSE`

### Test cases

- Check behavior on NaNs in `y_true` / `y_pred` / both
- Check behavior on all NaNs

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
See #512

## Closing issues
Closes #512.

### CODE REVIEW

File: etna/libs/sklearn/metrics.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+"""
```
Comment (d-a-bunin) [inline etna/libs/sklearn/metrics.py:1]: I'm not sure if we should add this into `etna.libs`, but it seems like change a very small and we should give a link to a license of sklearn somewhere.

File: etna/libs/sklearn/metrics.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+# Note: Copied from scikit-learn repository (https://github.com/scikit-learn/scikit-learn/blob/1.0.2/sklearn/metrics/_regression.py#L378)
```
Comment (d-a-bunin) [inline etna/libs/sklearn/metrics.py:32]: I'm not sure if we should use some other version. I took version from `poetry.lock`.

File: etna/metrics/functional_metrics.py [hunk 3]
```diff
@@ -41,6 +41,45 @@ def _get_axis_by_multioutput(multioutput: str) -> Optional[int]:
         assert_never(multioutput_enum)
 
 
+def mse_with_missing_handling(y_true: ArrayLike, y_pred: ArrayLike, multioutput: str = "joint") -> ArrayLike:
```
Comment (d-a-bunin) [inline etna/metrics/functional_metrics.py:44]: We should discuss naming and the fact that sklearn's `mse` is in our `etna/metrics/__init__.py`.

File: etna/metrics/base.py [hunk 4]
```diff
@@ -128,7 +128,7 @@ class Metric(AbstractMetric, BaseMixin):
     def __init__(
         self,
         metric_fn: MetricFunction,
-        mode: str = MetricAggregationMode.per_segment,
+        mode: str = MetricAggregationMode.per_segment.value,
```
Comment (brsnw250) [inline etna/metrics/base.py:131]: We should consider setting here a plain string instead of an enum value

File: etna/metrics/functional_metrics.py (hunk 3 above)
Comment (brsnw250) [reply etna/metrics/functional_metrics.py:45]: We can name it `mean_squared_error`. Mentioning missing handling in the name seems a bit excessive. Better to leave this clarification to the documentation.

File: etna/metrics/functional_metrics.py [hunk 5]
```diff
@@ -41,6 +42,52 @@ def _get_axis_by_multioutput(multioutput: str) -> Optional[int]:
...
+    """Mean squared error with missing values handling.
+
+    `Wikipedia entry on the Mean squared error
+    <https://en.wikipedia.org/wiki/Mean_squared_error>`_
```
Comment (brsnw250) [inline etna/metrics/functional_metrics.py:49]: I don't think it is a good idea to reference wikipedia. Better to replace with a more reputable source (e.g. Hyndman Forecasting) or to remove completely.

File: etna/metrics/functional_metrics.py [hunk 6]
```diff
@@ -41,6 +42,52 @@ def _get_axis_by_multioutput(multioutput: str) -> Optional[int]:
...
+
+    The nans are ignored during computation.
```
Comment (brsnw250) [inline etna/metrics/functional_metrics.py:51]: It would be helpful to also note what will be returned if all nans in the segment

**Review by brsnw250 — CHANGES_REQUESTED** [review]

File: etna/metrics/base.py (hunk 4 above)
Comment (d-a-bunin) [reply etna/metrics/base.py:131]: You suggest using just "per-segment"?

File: etna/metrics/functional_metrics.py (hunk 5 above)
Comment (d-a-bunin) [reply etna/metrics/functional_metrics.py:49]: Here I just repeated that we have done with `mape` and `smape`. I could replace link in those places too.

File: etna/metrics/functional_metrics.py (hunk 3 above)
Comment (d-a-bunin) [reply etna/metrics/functional_metrics.py:45]: If we have both `mse` from sklearn and our `mean_squared_error` it could be confusing.

We could remove sklearn's version, we haven't mentioned it in our public documentation (but have it in `etna/metrics/__init__.py`).

I'm also a bit of worried that by replacing sklearn's mse with our mse we changed list of available kwargs.
